src/main/resources/migreringer/V1.0_0023.sql

```diff
@@ -0,0 +1,12 @@
+CREATE INDEX ON oppgave((data->>'aktiv'));
+CREATE INDEX ON oppgave((data->>'system'));
+CREATE INDEX ON oppgave((data->>'kode6'));
+CREATE INDEX ON oppgave((data->'behandlingType'->>'kode'));
+CREATE INDEX ON oppgave((data->'fagsakYtelseType'->>'kode'));
+CREATE INDEX ON oppgave((data->'behandlingStatus'->>'kode'));
+CREATE INDEX ON oppgave((data->>'aktorId'));
+CREATE INDEX ON oppgave((data->>'pleietrengendeAktørId'));
+CREATE INDEX ON oppgave((data->>'journalpostId'));
+
+CREATE INDEX ON oppgave USING GIN (data);
```

Comment:
hva betyr dette? :)

Reply:
Det gjør at databasen bygger et tre som den bruker for binærsøk på de variablene. Alternativet er at den må åpne hele json og se etter variabelen for hver rad. 
GIN har jeg ikke brukt før, men skal indeksere alle nøklene i json automatisk, men vi må bygge opp sql-spørringene på en litt annen måte enn de er nå for å dra nytte av den.

Da jeg la på indekser gikk svartiden på `oppgaveTjeneste.hentBeholdningAvOppgaverPerAntallDager()` fra 2800ms til 370ms

Reply:
Ved 10 000 oppgaver i db. Mistenker at det er betydelig flere i prod..

Reply:
så når vi tar i bruk `GIN` så kan vi fjerne de andre indeksene fra linje 1-9?

Reply:
Nei, ikke rett frem å bruke det. Vi må skrive select-spørringer med json, som dette `WHERE data @> ‘{“aktiv” :“true”};`
Tror ikke det er noe særlig verdi å endre de eksisterende spørringene, men kanskje ved opprettelse av nye, eller noen som skal sjekke mange variabler i json

Reply:
ja ser det. Kan hende "vanlig" indeks er enda raskere siden jeg tror det er hash oppslag, mens binær søk tar bitte litt tid. Men morsomme greier!